### PR TITLE
Issue134 seeding

### LINKED ISSUE #134: Implement controlled random number seeds for the environment
To achieve reproduceable random behavior, a seeding mechanism has to be implemented through any random number generator in the environment.

### PR DESCRIPTION
**Pull request by atra94**
This PR adds the possibility to seed gym-electric-motor environments. The seeding enables controlled random behavior and therefore, a better comparability between multiple agents. They can be tested on exactly the same test reference trajectories, same initial state values after reset etc. 

Documentation in code and the link to the Sphinx API Reference are done. Furthermore, a small documentary section in the GEM Cookbook has been added. 

Unit tests and integration tests in the environment are available.

Closes #134

### CODE REVIEW

**Review by max-schenke — COMMENTED** [review]
Thank you for this work!

I will run a functional test ASAP, the code looks good.

As a general demand for future pull requests: I would like it if content and feature additions like
the seeding functionality would not be contained in the same pull request with such a lot of code reformatting.
Of course it is very welcome if code is restructured to be more readable, but checking the pull request for the relevant parts (that really change the code) is much more time consuming if this many files are affected. This is no criticism but that way we may be able to accelerate code review :)

File: gym_electric_motor/core.py [hunk 1]
```diff
@@ -21,8 +21,11 @@
 import numpy as np
 from gym.spaces import Box
 
+import gym_electric_motor
 from .utils import instantiate
+from .random_component import RandomComponent
 from .constraints import Constraint, LimitConstraint
+import gym_electric_motor as gem
```
[the remark this answers is not in the document]
Comment (atra94) [reply gym_electric_motor/core.py]: You are right. Probably PyCharm automatically added line 24. I removed it.

File: gym_electric_motor/physical_systems/physical_systems.py [hunk 2]
```diff
@@ -91,6 +94,11 @@ def __init__(self, converter, motor, load, supply, ode_solver, noise_generator=N
         self._system_eq_placeholder = None
         self._motor_deriv_size = None
         self._load_deriv_size = None
+        components = [
+            self._converter, self._electrical_motor, self._mechanical_load, self._ode_solver, self._supply,
+            self._noise_generator
+        ]
```
[the remark this answers is not in the document]
Comment (atra94) [reply gym_electric_motor/physical_systems/physical_systems.py]: It is only necessary that the order is the same for two seedings of the environment. The exact order does not matter. However, I changed the order such that it models more or less the calculation flow. This seemed to me the only "obvious" way to arrange the components.

File: gym_electric_motor/reference_generators/laplace_process_reference_generator.py [hunk 3]
```diff
@@ -17,12 +18,12 @@ def __init__(self, sigma_range=(1e-3, 1e-1), **kwargs):
             sigma_range(Tuple(float,float)): Lower and Upper limit for the sigma-parameter of the WienerProcess.
             kwargs: Further arguments to pass to SubepisodedReferenceGenerator
         """
-        super().__init__(**kwargs)
+        SubepisodedReferenceGenerator.__init__(**kwargs)
         self._sigma_range = sigma_range
 
     def _reset_reference(self):
         self._current_sigma = 10 ** self._get_current_value(np.log10(self._sigma_range))
```
[the remark this answers is not in the document]
Comment (atra94) [reply gym_electric_motor/reference_generators/laplace_process_reference_generator.py:25]: This is the random drawing of the uniformly distributed values from a logarithmic x-scale.
The logarithmic scale lead to a better distribution of random values.

File: tests/test_physical_systems/test_noise_generator.py [hunk 4]
```diff
@@ -3,79 +3,9 @@
...
-g_noise_levels = np.array([1, 10, 0.5, 0, 1])
-g_state_variables = ['omega', 'torque', 'i', 'u', 'u_sup']
-g_signal_powers = np.array([1, 2, 3, 4, 5])
-
-
-def test_gaussian_white_noise_generator():
-    """
-    This function test the Gaussian White Noise Generator:
-    Following functions are included:
-    NoiseGenerator
```
[the remark this answers is not in the document]
Comment (atra94) [reply tests/test_physical_systems/test_noise_generator.py:18]: Sorry, I cannot follow your argumentation.

There are still multiple tests in the file. The intention of the PR is to use the numpy random generators and replace the "simple" numpy random functions to achieve a seedable enironment. A random generator is only part of the noise generator. I think you mix something up here.

**Review by wallscheid — APPROVED** [review]
Nice work - thanks to @atra94 !

Only one minor comment: The tests for the ac and dc env. seedings are rather uncommented so far. I think it is not of highest priority, but some general explanation on the conducted tests could be helpful.

**Pull request merged**